L1Trigger/L1THGCal/src/HGCalTriggerTowerGeometryHelper.cc

```diff
@@ -120,18 +202,72 @@ unsigned short HGCalTriggerTowerGeometryHelper::getTriggerTowerFromEtaPhi(const
   return l1t::HGCalTowerID(doNose_, zside, bin_eta, bin_phi).rawId();
 }
 
-unsigned short HGCalTriggerTowerGeometryHelper::getTriggerTower(const l1t::HGCalTriggerCell& thecell) const {
+std::unordered_map<unsigned short, float> HGCalTriggerTowerGeometryHelper::getTriggerTower(const l1t::HGCalTriggerCell& thecell) const {
+  std::unordered_map<unsigned short, float> towerIDandShares = {}; 
   unsigned int trigger_cell_id = thecell.detId();
   // NOTE: if the TC is not found in the map than it is mapped via eta-phi coords.
   // this can be considered dangerous (silent failure of the map) but it actually allows to save
   // memory mapping explicitly only what is actually needed
   auto tower_id_itr = cells_to_trigger_towers_.find(trigger_cell_id);
-  if (tower_id_itr != cells_to_trigger_towers_.end())
-    return tower_id_itr->second;
-
-  return getTriggerTowerFromEtaPhi(thecell.position().eta(), thecell.position().phi());
+  if (tower_id_itr != cells_to_trigger_towers_.end()){
+    towerIDandShares.insert({tower_id_itr->second, 1.0});
+    return towerIDandShares;
+  }
+  towerIDandShares.insert({getTriggerTowerFromEtaPhi(thecell.position().eta(), thecell.position().phi()), 1.0});
+  return towerIDandShares;
 }
 
-unsigned short HGCalTriggerTowerGeometryHelper::getTriggerTower(const l1t::HGCalTriggerSums& thesum) const {
-  return getTriggerTowerFromEtaPhi(thesum.position().eta(), thesum.position().phi());
+std::unordered_map<unsigned short, float> HGCalTriggerTowerGeometryHelper::getTriggerTower(const l1t::HGCalTriggerSums& thesum) const {
+  std::unordered_map<unsigned short, float> towerIDandShares = {}; 
+  if(!splitModuleSum_){
+    towerIDandShares.insert({getTriggerTowerFromEtaPhi(thesum.position().eta(), thesum.position().phi()), 1.0});
+    return towerIDandShares;
+  } else{
+    HGCalModuleDetId detid(thesum.detId());
+    int moduleU = detid.moduleU();
+    int moduleV = detid.moduleV();
+    int layer = detid.layer();
+    int sector = detid.sector();
+    int zside = detid.zside();
+    int subdet = -1;
+    int splitDivisor = -1;
```

Comment:
Default values different than -1 (e.g. using values corresponding to EE) should probably be used, otherwise it could possibly do some strange things, in particular for the HFNose.
And would be good to fallback to the eta/phi behaviour in case it falls in none of the three expected cases. This way HFNose modules will fallback to eta/phi towers automatically.